src/workers/fetchers/curve/CurvePriceFetcher.ts

```diff
@@ -47,9 +49,10 @@ export class CurvePriceFetcher extends BaseFetcher<CurveWorkerConfiguration> {
       const historyFile: HistoryFile = JSON.parse(fs.readFileSync(historyFileName, 'utf-8'));
       startBlock = historyFile.lastBlockFetched + 1;
     } else {
-      startBlock =
-        (await Web3Utils.GetContractCreationBlockNumber(curvePricePairConfiguration.poolAddress, this.workerName)) +
-        100_000;
+      startBlock = curvePricePairConfiguration.minBlock
+        ? curvePricePairConfiguration.minBlock
+        : (await Web3Utils.GetContractCreationBlockNumber(curvePricePairConfiguration.poolAddress, this.workerName)) +
+          100_000;
     }
```

Comment:
Here it means that for BSC we NEED to have minBlock in every config because otherwise it will fetch using etherscan which wont work.

The valid feature would be to have a switch on the chainId from the web3Provider and select etherscan for mainnet, bscscan for bsc etc... But I'm not sure if bscscan provides the same api as etherscan when it comes to contracts